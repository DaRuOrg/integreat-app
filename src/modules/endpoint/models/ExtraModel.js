class ExtraModel {
  constructor ({name, type, url}) {
    this._name = name
    this._type = type
    this._url = url
  }

  get name () {
    return this._name
  }

  get type () {
    return this._type
  }

  get url () {
    return this._url
  }
}

export default ExtraModel
