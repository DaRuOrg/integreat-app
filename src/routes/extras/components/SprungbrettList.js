import React from 'react'
import PropTypes from 'prop-types'

import SprungbrettJobModel from 'modules/endpoint/models/SprungbrettJobModel'

import Caption from '../../../modules/common/components/Caption'
import SprungbrettListItem from './SprungbrettListItem'

class SprungbrettList extends React.Component {
  static propTypes = {
    jobs: PropTypes.arrayOf(PropTypes.instanceOf(SprungbrettJobModel)).isRequired,
    title: PropTypes.string
  }

  getListItems () {
    return this.props.jobs.map(job => <SprungbrettListItem key={job.title()} job={job} />)
  }

  render () {
    return (
      <div>
        {this.props.title && <Caption title={this.props.title} />}
        {this.getListItems()}
      </div>
    )
  }
}

export default SprungbrettList
