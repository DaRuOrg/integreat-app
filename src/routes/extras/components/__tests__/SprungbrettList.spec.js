import React from 'react'
import { shallow } from 'enzyme'
import SprungbrettList from '../SprungbrettList'
import ExtraModel from '../../../../modules/endpoint/models/ExtraModel'

describe('SprungbrettList', () => {
  const extras = [
    new ExtraModel({type: 'ige-sbt', path: 'path to fetch jobs from', name: 'Sprungbrett', thumbnail: 'xy'}),
    new ExtraModel({type: 'ige-ilb', path: 'ihk-jobborese.com', name: 'Jobboerse', thumbnail: 'xy'}),
    new ExtraModel({type: 'ige-ipb', path: 'ihk-pratkitkumsboerse.com', name: 'Praktikumsboerse', thumbnail: 'xy'})
  ]
  it('should match snapshot', () => {
    const wrapper = shallow(<SprungbrettList jobs={extras} />)
    expect(wrapper).toMatchSnapshot()
  })
})
