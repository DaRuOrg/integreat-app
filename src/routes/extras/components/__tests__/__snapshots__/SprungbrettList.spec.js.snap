// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SprungbrettList should match snapshot 1`] = `
<div
  className="list"
>
  <SprungbrettListItem
    job={
      ExtraModel {
        "_name": "Sprungbrett",
        "_path": "path to fetch jobs from",
        "_thumbnail": "xy",
        "_type": "ige-sbt",
      }
    }
  />
  <SprungbrettListItem
    job={
      ExtraModel {
        "_name": "Jobboerse",
        "_path": "ihk-jobborese.com",
        "_thumbnail": "xy",
        "_type": "ige-ilb",
      }
    }
  />
  <SprungbrettListItem
    job={
      ExtraModel {
        "_name": "Praktikumsboerse",
        "_path": "ihk-pratkitkumsboerse.com",
        "_thumbnail": "xy",
        "_type": "ige-ipb",
      }
    }
  />
</div>
`;
